doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/config.md

```diff
@@ -383,3 +440,36 @@ environment variables, for debugging and development purposes.
 These fields should be formatted following the
 [Kuberentes API](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#environment-variables)
 for container environment variables.
+
+(kuberay-config-miscellaneous)=
+## Pod and container lifecyle: preStop hooks and initContainers
+There are two pieces of pod configuration that should always be included
+in the RayCluster CR. Future versions of KubeRay may configure these elements automatically.
+
+## initContainer
```

Comment:
Oh wow, I didn't realize this -- shouldn't we just fix this in KubeRay and make sure the head service is created before the workers? :)

Reply:
It looks like the service is actually created before the worker.
https://github.com/ray-project/kuberay/blob/76f96bf5a05651611f9a8734c34c2a7f23f54b54/ray-operator/controllers/ray/raycluster_controller.go#L196-L211
I imagine K8s's internal service discovery is not instantaneous, though.